File: CS-Examples/07_DocumentOperation/RemoveVariables.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace RemoveVariables
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
           // Create a new Document object
			Document document = new Document();

			// Load a Word document from the specified file path
			document.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_Docx_6.docx");

			// Remove the variable named "A1" from the document's Variables collection
			document.Variables.Remove("A1");

			// Set the IsUpdateFields property of the document to true, enabling field updates
			document.IsUpdateFields = true;

			// Specify the file name for the saved document
			string result = "Result-RemoveVariables.docx";

			// Save the document to a file in DOCX format (using Word 2013 format)
			document.SaveToFile(result, FileFormat.Docx2013);

			// Release the resources used by the document
			document.Dispose();

            //Launch the MS Word file.
            WordDocViewer(result);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/07_DocumentOperation/RemoveVariables.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace RemoveVariables
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new instance of the Document class
			Dim document As New Document()

			' Load the specified document file
			document.LoadFromFile("..\..\..\..\..\..\..\Data\Template_Docx_6.docx")

			' Remove the variable with the specified name from the document
			document.Variables.Remove("A1")

			' Set the IsUpdateFields property to true to update fields when saving the document
			document.IsUpdateFields = True

			' Specify the file path for the output result
			Dim result As String = "Result-RemoveVariables.docx"

			' Save the document to a file with the specified file format
			document.SaveToFile(result, FileFormat.Docx2013)

			' Dispose of the document object
			document.Dispose()

			'Launch the MS Word file.
			WordDocViewer(result)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
